controllers/bsl.go

```diff
@@ -134,10 +134,27 @@ func (r *DPAReconciler) ReconcileBackupStorageLocations(log logr.Logger) (bool,
 				}
 				bsl.Spec.Credential = bslSpec.CloudStorage.Credential
 				bsl.Spec.Default = bslSpec.CloudStorage.Default
-				bsl.Spec.Provider = AWSProvider
 				bsl.Spec.ObjectStorage = &velerov1.ObjectStorageLocation{
 					Bucket: bucket.Spec.Name,
 				}
+				switch bucket.Spec.Provider {
+				case oadpv1alpha1.AWSBucketProvider:
+					{
```

Comment:
Just a NIT I don't think you actually need `{}` here for a case statement. But it brought me back to my C days :)